buffer/src/main/java/io/netty/buffer/AdaptivePoolingAllocator.java

```diff
@@ -302,6 +302,32 @@ private boolean offerToQueue(Chunk buffer) {
         return centralQueue.offer(buffer);
     }
 
+    // Ensure that we release all previous pooled resources when this object is finalized. This is needed as otherwise
+    // we might end up with leaks. While these leaks are usually harmless in reality it would still at least be
+    // very confusing for users.
+    @Override
+    protected void finalize() throws Throwable {
```

Comment:
I'm thinking it might be a good idea to place the finalizer in a separate object, similar to what we do in the PooledByteBufAllocator.

Reply:
I will think about it and might do a followup